generated/version.h

```diff
@@ -0,0 +1,19 @@
+#ifndef NIDEVICE_GRPC_VERSION_H
+#define NIDEVICE_GRPC_VERSION_H
+
+#include <string>
+
+namespace nidevice_grpc {
+
+std::string const NIDEVICE_GRPC_MAJORNUMBER = "2";
```

Comment:
Defining static STL objects in a header file duplicates them in every translation unit that includes the header.

Reply:
Option 1: use C strings

version.h:
```
const char* const NIDEVICE_GRPC_MAJORNUMBER = "2";
```

Option 2: use extern STL strings

version.h:
```
extern const std::string NIDEVICE_GRPC_MAJORNUMBER;
```

version.cpp
```
const std::string NIDEVICE_GRPC_MAJORNUMBER = "2";
```

Reply:
`constexpr` C string pointers are also fine:
```
constexpr const char* NIDEVICE_GRPC_MAJORNUMBER = "2";
```

Reply:
Why the `static`? Since it's a constexpr, won't we just evaluate it at compile time and then not need it anymore?

Reply:
For a variable at file/namespace scope, [`static` means "internal linkage"](https://en.cppreference.com/w/cpp/language/storage_duration), so each translation unit (.cpp file) basically gets a copy, preventing name collisions between translation units. I don't know if it's necessary for `constexpr`. In C++, `const` variables are static by default and non-`const` variables are global by default. (AFAIK C has different rules.)

`constexpr` means the variable can be used at compile time, not that the variable cannot exist at run time.